source/node/Group.js

```diff
@@ -130,21 +126,26 @@ class Group {
 
     /**
      * Get an attribute
-     * @param {string} attributeName The name of the attribute
-     * @returns {string|undefined} Returns the attribute or undefined if not found
+     * @param {String=} attributeName The name of the attribute. If none provided
+     *  the entire attributes object is returned.
+     * @returns {String|undefined|Object} Returns the attribute or undefined if not found.
+     *  If no attribute name is provided an object containing all attributes is returned.
      * @memberof Group
      */
     getAttribute(attributeName) {
-        debug("fetch attribute");
-        const raw = this._getRemoteObject();
-        return raw.attributes && raw.attributes.hasOwnProperty(attributeName)
-            ? raw.attributes[attributeName]
-            : undefined;
+        const attributes = this._getRemoteObject().attributes || {};
+        if (typeof attributeName === "undefined") {
```

Comment:
Would it be possible to reverse this logic and check if `attributeName` is string? Because this would fail if for example numbers or other types are passed in that are not "undefined". Such as null.

Reply:
Well, I'm not sure if that'd be the best case. The edge case here is if `attributeName` is undefined. All other values should be assumed to be keys, but this one unique value should trigger special functionality. Obviously object values and numbers, for example, are also invalid, but we're trusting that people aren't sending those...

Reply:
I was mostly thinking about null but you’re right.